### PR TITLE
Update TestDrive and TestRegistry

### LINKED ISSUE #94: TestDrive should say clearly it setups only for the top-level block
TestDrive docs should make it clear that test drive is setup just once for the whole top level block, no matter if it is called Describe or Context, and is not setup for every block in the top level block.

Original issue: 

The documentation could be clearer about this. (It explicitly says Context block when it's actually any nested Context or Describe block.)

_Originally posted by @Axiometry in https://github.com/pester/Pester/issues/1837#issuecomment-773204331_

### PR DESCRIPTION
**Pull request by fflaten**
Updates docs regarding scoping of TestDrive and TestRegistry

Fix #94

### CODE REVIEW

File: docs/usage/testdrive.mdx [hunk 1]
```diff
@@ -4,19 +4,28 @@ title: Isolating File Operations using the TestDrive
...
 
 ## Scoping
 
-Basic scoping rules are implemented for the TestDrive. A clean TestDrive is created for every [Describe](../commands/Describe) and all the files created are available in the whole [Describe](../commands/Describe) scope. If the [Context](../commands/Context) keyword is also used the state of the TestDrive is recorded before moving into the Context block. Inside the [Context](../commands/Context) block the files from the [Describe](../commands/Describe) scope are available for reading and modification. You can move them around and create new ones as well.
+Basic scoping rules are implemented for the TestDrive:
 
-Once the [Context](../commands/Context) block is finished all the files created inside that block are deleted, leaving only the files created in the Describe block. When the Describe block is finished all contents of the TestDrive are discarded.
+1. A clean TestDrive is created per container (test-file or scriptblock) on entry to the first top-level block (Describe or Context) in that container.
```
Comment (nohwnd) [inline docs/usage/testdrive.mdx:15]: I think this is for each top-level block in container. Not whole container.

Comment (fflaten) [reply docs/usage/testdrive.mdx:15]: Might have to rephrase. 
The path for the TestDrive-root is the same for the whole container. The content however is cleared between each top-level block, as part of "exiting block" cleanup on any level. I tried to cover that part in a later step.

Comment (nohwnd) [reply docs/usage/testdrive.mdx:15]: Ok. Let me try it first.

Comment (nohwnd) [reply docs/usage/testdrive.mdx:15]: Sorry you are right. This probably changed with the introduction of root block. Root is now acting as the top-level block and all describes in the file share the same drive. The content is cleared, unless you add it in the top-level BeforeAll. In that case it stays. Which is correct, it is still the same logical behavior like when returning from any other child describe. The items that were created above the current level are still present, unless you removed them yourself in the child.

**Pull request merged**